### ISSUE TITLE
Extend Predicates with exists/contains method for Collections

### ISSUE THREAD

**Issue by lubos-bistak** [javaslang/javaslang#1470]
# Application Code

```
private Result calculateResult(final Seq<Result> results) {
    return Match(results).of(
      Case(exists(Result::isCritical), CRITICAL),
      Case(exists(Result::isWarning), WARNING),
      Case($(), OK)
    );
}
```
# Javaslang Code

```
public static <T> Predicate<Traversable<T>> exists(final Predicate<T> predicate) {
    Objects.requireNonNull(predicate, "predicate is null");
    return obj -> obj.exists(predicate);
}
```

What do you think?

**Comment by danieldietrich** [comment]
Sounds good! We should do the following:

### Use Value instead of Traversable

```java
<T> Predicate<Value<T>> exists(Predicate<? super T> predicate)
```

### Use `? super T` in the argument

```java
<T> Predicate<Value<T>> exists(Predicate<? super T> predicate)
```

### Also add a `forAll`:

```java
public static <T> Predicate<Value<T>> forAll(Predicate<? super T> predicate) {
    Objects.requireNonNull(predicate, "predicate is null");
    return obj -> obj.forAll(predicate);
}
```

(I've also omitted the `final` keyword for method parameters)

**Comment by danieldietrich** [comment]
Fixed with #1920
